proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,257 @@
+
+More formally, a function call crosses an ownership boundary when it is
+important that the only information communicated to the function involves data
+that is explicitly passed as arguments to the call, and that there is no
+possibility that the callee can side-effect the caller's state (by, for example,
+overwriting areas of linear memory). Nor can the caller side-effect the callee's
+state other than by invoking one of the callee's publicly exported APIs.
+
+>Note that the 'other' aspects of ownership -- such as who is responsible for
+>the code -- are also important but beyond the scope of this note.
+
+This limiting of trust is a vital enabler for the API ecosystem: if
+functionalities can be published and consumed with limited trust it makes it
+more likely that those same functionalities will be published and consumed.
+
+It is a goal of the Interface Types proposal that this limited trust can be
+supported by proving that an imported module cannot side-effect the importing
+module (and vice versa). This proof taking the form of a simple validation step
+performed during module instantiation.
+
+### Interoperability via Function Calls
+
+Interoperability between different systems is a hallmark of many if not most
+modern applications. However, one of the differentiating aspects of
+interoperability as it applies to webAssembly is the role of the function
+call. Unlike interoperability expressed at the level of web services (say),
+using a foreign capability in webAssembly is most naturally expressed as a
+function call.
+
+I.e., invoking capabilities does not naturally imply sending messages,
+serializing data, network connections, or even crossing a process boundary: the
+invoked capability may even be present in the same thread.
+
+This context informs much of the style of how interoperability is expressed in
+the Interface Types proposal. In particular, this proposal does not include a
+serialization format.[^Although there is a binary encoding, its role is to
+support the publishing and loading of webAssembly modules; not to support
+function interaction.] Indeed, it is expected that no serialization is required
+when invoking a function via Interface Types' adapters.
+
+### Language Specific Module Systems
+
+Interface Types are _not_ intended to replacement language specific module
+systems. For example, C/C++ modules would continue to be linked together into
+single webAssembly modules. 
+
+## WebIDL
+
+[Web IDL][WebIDL] is clearly an important IDL for Web APIs. And, since the web
+platform is a crucial domain for webAssembly, it seems necessary to address the
+question of "why not WebIDL" for describing APIs in Interface Types.
+
+The case against WebIDL is comprised of three principal arguments:
+
+* WebIDL is too strongly aligned to JavaScript;
+* WebIDL has features that are not relevant in the context of webAssembly; and
+* WebIDL does not support certain necessary features.
+
+### Javascript
```

Comment:
Another capitalization note: "JavaScript" is the accepted capitalization for this term (as used above).